
Author: rmh
Description:  Move _PATH_LOG to bits/syslog-path.h (this way we can override it)
Status: merged (see http://sources.redhat.com/bugzilla/show_bug.cgi?id=2172)

diff -Nur misc/Makefile misc/Makefile
--- misc/Makefile	2005-07-30 01:26:18.000000000 +0200
+++ misc/Makefile	2006-02-24 23:42:01.000000000 +0100
@@ -25,7 +25,7 @@
 headers	:= sys/uio.h bits/uio.h sys/ioctl.h bits/ioctls.h bits/ioctl-types.h \
 	   sys/ptrace.h sys/file.h sys/dir.h sys/cdefs.h \
 	   ar.h a.out.h libgen.h stab.h bits/stab.def sgtty.h \
-	   ttyent.h syslog.h sys/syslog.h paths.h sys/reboot.h \
+	   ttyent.h syslog.h sys/syslog.h bits/syslog-path.h paths.h sys/reboot.h \
 	   sys/mman.h sys/param.h fstab.h mntent.h search.h err.h error.h \
 	   sys/queue.h sysexits.h syscall.h sys/syscall.h sys/swap.h \
 	   sys/select.h ustat.h sys/ustat.h bits/ustat.h sys/sysinfo.h \
diff -Nur misc/sys/syslog.h misc/sys/syslog.h
--- misc/sys/syslog.h	2003-08-30 11:27:50.000000000 +0200
+++ misc/sys/syslog.h	2006-02-24 23:41:39.000000000 +0100
@@ -37,7 +37,7 @@
 #include <stdarg.h>
 
 
-#define	_PATH_LOG	"/dev/log"
+#include <bits/syslog-path.h>
 
 /*
  * priorities/facilities are encoded into a single 32-bit quantity, where the
diff -Nur sysdeps/generic/bits/syslog-path.h sysdeps/generic/bits/syslog-path.h
--- sysdeps/generic/bits/syslog-path.h	1970-01-01 01:00:00.000000000 +0100
+++ sysdeps/generic/bits/syslog-path.h	2006-02-24 23:41:39.000000000 +0100
@@ -0,0 +1,29 @@
+/* <bits/syslog-path.h> -- _PATH_LOG definition
+   Copyright (C) 2006 Free Software Foundation, Inc.
+   This file is part of the GNU C Library.
+
+   The GNU C Library is free software; you can redistribute it and/or
+   modify it under the terms of the GNU Lesser General Public
+   License as published by the Free Software Foundation; either
+   version 2.1 of the License, or (at your option) any later version.
+
+   The GNU C Library is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+   Lesser General Public License for more details.
+
+   You should have received a copy of the GNU Lesser General Public
+   License along with the GNU C Library; if not, write to the Free
+   Software Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA
+   02111-1307 USA.  */
+
+#ifndef _SYS_SYSLOG_H
+# error "Never include this file directly.  Use <sys/syslog.h> instead"
+#endif
+
+#ifndef _BITS_SYSLOG_PATH_H
+#define _BITS_SYSLOG_PATH_H 1
+
+#define	_PATH_LOG	"/dev/log"
+
+#endif /* bits/syslog-path.h */
